changelog/8-1-2-Release.md

```diff
@@ -0,0 +1,1109 @@
+
+**Example**:
+
+{% tabs %}
+{% tab title="Before" %}
+```java
+package org.openrewrite.kubernetes.resource;
+
+import lombok.EqualsAndHashCode;
+import lombok.Value;
+import org.openrewrite.*;
+import org.openrewrite.yaml.YamlIsoVisitor;
+import org.openrewrite.yaml.tree.Yaml;
+
+@Value
+@EqualsAndHashCode(callSuper = true)
+public class FindExceedsResourceRatio extends Recipe {
+    @Override
+    public String getDisplayName() {
+        return "Find exceeds resource ratio";
+    }
+
+    @Override
+    protected TreeVisitor<?, ExecutionContext> getVisitor() {
+        return new YamlIsoVisitor<ExecutionContext>() {
+            @Override
+            public Yaml.Mapping.Entry visitMappingEntry(Yaml.Mapping.Entry entry, ExecutionContext ctx) {
+                Yaml.Mapping.Entry e = super.visitMappingEntry(entry, ctx);
+                return e.withMarkers(e.getMarkers().searchResult("foo"));
+            }
+        };
+    }
+}
+```
+{% endtab %}
+
+{% tab title="After" %}
+```java
+package org.openrewrite.kubernetes.resource;
+
+import lombok.EqualsAndHashCode;
+import lombok.Value;
+import org.openrewrite.*;
+import org.openrewrite.marker.SearchResult;
+import org.openrewrite.yaml.YamlIsoVisitor;
+import org.openrewrite.yaml.tree.Yaml;
+
+@Value
+@EqualsAndHashCode(callSuper = true)
+public class FindExceedsResourceRatio extends Recipe {
+    @Override
+    public String getDisplayName() {
+        return "Find exceeds resource ratio";
+    }
+
+    @Override
+    protected TreeVisitor<?, ExecutionContext> getVisitor() {
+        return new YamlIsoVisitor<ExecutionContext>() {
+            @Override
+            public Yaml.Mapping.Entry visitMappingEntry(Yaml.Mapping.Entry entry, ExecutionContext ctx) {
+                Yaml.Mapping.Entry e = super.visitMappingEntry(entry, ctx);
+                return SearchResult.found(e, "foo");
+            }
+        };
+    }
+}
+```
+{% endtab %}
+{% endtabs %}
+
+### The doNext method has been removed
+
+The [org.openrewrite.Recipe doNext(..)](https://github.com/openrewrite/rewrite/blob/v7.40.8/rewrite-core/src/main/java/org/openrewrite/Recipe.java#L306-L313) method has been removed. If a recipe won't make any changes, this should be replaced with a [ScanningRecipe](#new-approach-to-recipe-visiting). If it does make changes, it should use [TreeVisitor#doAfterVisit()](https://github.com/openrewrite/rewrite/blob/v7.40.8/rewrite-core/src/main/java/org/openrewrite/TreeVisitor.java#L146).
+
+TODO: This migration recipe does not provide enough context. It links to a non-existant guide: https://github.com/openrewrite/rewrite/blob/main/rewrite-java-test/src/test/java/org/openrewrite/java/recipes/MigrateRecipeToRewrite8Test.java#L485
```

Comment:
To help make reviewing easier - pointing out the TODOs that I feel like still need work.

Reply:
Should I call out that `doAfterVisit(Recipe)` has been removed then in combination with this? 

Also - I'm not sure I follow what you're saying. Were people using `doNext(Recipe)` to run the recipe they're currently in? Why would that be the case?

Reply:
> Should I call out that doAfterVisit(Recipe) has been removed then in combination with this?

Yes, I think so.

> Also - I'm not sure I follow what you're saying. Were people using doNext(Recipe) to run the recipe they're currently in? Why would that be the case?

Not to run the current recipe in `doNext(Recipe)`, but a different recipe